java/arcs/core/entity/EntityBase.kt

```diff
@@ -36,7 +36,8 @@ open class EntityBase(
     entityId: String? = null,
     creationTimestamp: Long = UNINITIALIZED_TIMESTAMP,
     expirationTimestamp: Long = UNINITIALIZED_TIMESTAMP,
-    private val isInlineEntity: Boolean = false
+    private val isInlineEntity: Boolean = false,
+    private val restrictedEntitySchema: Schema? = null
```

Comment:
When will this be null and when will it be present?

Reply:
Maybe a different question:

Entities are created by code running inside the particle. While inside the particle, the type of a handle to which an entity will be written is not yet known. So I am not sure that EntityBase should have a `restrictedEntitySchema` field. WDYT?

Reply:
Instead of performing field dropping inside the entity (which does not know where it will be written), it may be better to do it inside a EntityStorageAdapter.storableToReferencable, as EntityStorageAdapter is created for a specific handle, so it is easy to enhance it with the type of the handle, which is the restricted type.

Reply:
Isn't there a new instance of the entity for each particle handle?

Reply:
:+1: (see above)